zipline/gens/tradesimulation.py

```diff
@@ -256,13 +299,13 @@ def _process_snapshot(self, dt, snapshot, instant_fill):
             # Now that handle_data has been called and orders have been placed,
             # process the event stream to fill user orders based on the events
             # from this snapshot.
-            for event in events_to_be_processed:
-                for txn, order in blotter_process_trade(event):
+            for trade in events_to_be_processed:
+                for txn, order in blotter_process_trade(trade):
```

Comment:
This is a quirk of the implementation to support the `instant_fill` behavior, which is provided as an option to have daily mode execute on the same day. However, I don't know how much use that behavior has.

Depending on the value of the flag which is a simulation "constant", either this block, or the one above is executed.